### PR TITLE
Add Logger

### PR DESCRIPTION
**Pull request by BitSyndicate1**
Adds a logger for global loggin access with macros

### CODE REVIEW

File: vr-logger/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,322 @@
...
+    }
+}
+
+impl Logger {
+    /// Format view:
+    /// ```norust
+    /// "%t %T %c %f:%l %s"
+    /// "%r"
+    /// ```
+    pub fn new(path: PathBuf, format: &str) -> Result<Self, Error> {
```
Comment (anonymous123-code) [inline vr-logger/src/lib.rs:278]: We should enable to log to stdout

File: vr-logger/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,322 @@
...
+}
+
+const SECONDS_IN_A_DAY: u64 = SECONDS_IN_AN_HOUR * 24;
+const SECONDS_IN_AN_HOUR: u64 = SECONDS_IN_A_MINUTE * 60;
+const SECONDS_IN_A_MINUTE: u64 = 60;
+const DAYS_IN_A_YEAR: u64 = 365;
+
+const UNIX_EPOCH_YEAR: u64 = 1970;
+
+pub fn generate_utc_string() -> String {
```
Comment (anonymous123-code) [inline vr-logger/src/lib.rs:150]: Are there no inbuild methods/good libraries for this? (I want to stay off of time stuff as much as possible)

File: vr-threading/src/global_threadpool.rs [hunk 3]
```diff
@@ -0,0 +1,12 @@
+use std::sync::mpsc;
```
Comment (anonymous123-code) [inline vr-threading/src/global_threadpool.rs]: Is this being included intentional?

**Review by anonymous123-code — CHANGES_REQUESTED** [review]

File: vr-logger/src/lib.rs (hunk 2 above)
Comment (BitSyndicate1) [reply vr-logger/src/lib.rs:150]: In comparison to those libraries this is more than minimal, it is also correct

Comment (BitSyndicate1) [reply vr-logger/src/lib.rs:150]: 2. The only point where this is going to be used are the log messages

File: vr-logger/src/lib.rs (hunk 1 above)
Comment (BitSyndicate1) [reply vr-logger/src/lib.rs:278]: it logs to stdout and a file

File: vr-threading/src/global_threadpool.rs (hunk 3 above)
Comment (BitSyndicate1) [reply vr-threading/src/global_threadpool.rs]: Yes?

**Pull request merged**